scripts/publisherDVL1000.py

```diff
@@ -6,295 +6,354 @@
+        BottomXyzVel3Valid = theData["frames"][6]["inputs"][0]["lines"][2]["valid"]
+        BottomXyzVel4Valid = theData["frames"][6]["inputs"][0]["lines"][3]["valid"]
+
+        # XYZ FOM Variables
+        BottomXyzFomMin = theData["frames"][6]["inputs"][1]["min"]
+        BottomXyzFomMax = theData["frames"][6]["inputs"][1]["max"]
+        BottomXyzFomUnit = theData["frames"][6]["inputs"][1]["units"]
+        BottomXyzFom1Data = theData["frames"][6]["inputs"][1]["lines"][0]["data"][0]
+        BottomXyzFom2Data = theData["frames"][6]["inputs"][1]["lines"][1]["data"][0]
+        BottomXyzFom3Data = theData["frames"][6]["inputs"][1]["lines"][2]["data"][0]
+        BottomXyzFom4Data = theData["frames"][6]["inputs"][1]["lines"][3]["data"][0]
+        BottomXyzFom1Valid = theData["frames"][6]["inputs"][1]["lines"][0]["valid"]
+        BottomXyzFom2Valid = theData["frames"][6]["inputs"][1]["lines"][1]["valid"]
+        BottomXyzFom3Valid = theData["frames"][6]["inputs"][1]["lines"][2]["valid"]
+        BottomXyzFom4Valid = theData["frames"][6]["inputs"][1]["lines"][3]["valid"]
+
+        theDVL.header.stamp = rospy.Time.now()
+        theDVL.header.frame_id = "dvl_link"
+        theDVL.velocity.x = BottomXyzVel1Data
+        theDVL.velocity.y = BottomXyzVel2Data
+        if BottomXyzFom3Data > BottomXyzFom4Data:
+            theDVL.velocity.z = BottomXyzVel4Data
+            BottomXyzFomZbest = BottomXyzFom4Data
+        else:
+            theDVL.velocity.z = BottomXyzVel3Data
+            BottomXyzFomZbest = BottomXyzFom3Data
+
+        # Doing covariances
+        velocity_covariance = [
+            BottomXyzFom1Data * BottomXyzFom1Data,
+            unknown,
+            unknown,
+            unknown,
+            BottomXyzFom2Data * BottomXyzFom2Data,
+            unknown,
+            unknown,
+            unknown,
+            BottomXyzFomZbest * BottomXyzFomZbest,
+        ]
+
+        # Feeding message
+        theDVL.velocity_covariance = velocity_covariance
+        theDVL.altitude = (
+            BottomBeamDist1Data
+            + BottomBeamDist2Data
+            + BottomBeamDist3Data
+            + BottomBeamDist4Data
+        ) / 4
+
+        # Individual beam data
+        beam1 = theDVLBeam
+        beam1.range = BottomBeamDist1Data
+        beam1.range_covariance = 0.0001  # TODO: find accurate value
+        beam1.velocity = BottomBeamVel1Data
+        beam1.velocity_covariance = BottomBeamFom1Data * BottomBeamFom1Data
+        beam1.pose.header.stamp = rospy.Time.now()
+        beam1.pose.pose.position.x = 0.283
+        beam1.pose.pose.position.y = 0.283
+        beam1.pose.pose.position.z = 0
+        beam1.pose.pose.orientation.x = (
+            0.211  # Estimating values here. 25 degree tilt and 4 cm from origo
+        )
+        beam1.pose.pose.orientation.y = 0.211
+        beam1.pose.pose.orientation.z = -0.047
+        beam1.pose.pose.orientation.w = 0.953
+
+        beam2 = theDVLBeam
```

Comment:
How does this work in relation to object creation? This has bugged me from the previous version too, but never looked into it. It would seem more intuitive to directly create a DVLBeam() object for each beam#

Reply:
I haven't looked closer at that part yet, but I seems very strange to me as well. My first though was that this might cause overwrite issues, but since it seems to be working that's probably not the case. I think we should just remove `theDVLBeam` and replace it with normal object instantiation.